src/string_search.h

```diff
@@ -110,7 +110,14 @@ class StringSearch : private StringSearchBase {
   }
 
   size_t Search(Vector subject, size_t index) {
+#if (__GNUC__ >= 8) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
     return (this->*strategy_)(subject, index);
+#if (__GNUC__ >= 8) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
```

Comment:
IIUC the issue is not with `strategy_`, but with the arrays in the `StringSearchBase`, which are indeed uninitialized (they will be initialized before BoyerMoore search is started)

Reply:
@lundibundi Thanks, I'll try to take a closer look at this again this week.